nano/node/active_transactions.cpp

```diff
@@ -639,7 +639,12 @@ void nano::active_transactions::update_active_difficulty (std::unique_lock<std::
 	auto sum (std::accumulate (multipliers_cb.begin (), multipliers_cb.end (), double(0)));
 	auto difficulty = nano::difficulty::from_multiplier (sum / multipliers_cb.size (), node.network_params.network.publish_threshold);
 	assert (difficulty >= node.network_params.network.publish_threshold);
+	
+	bool notify_change = trended_active_difficulty != difficulty;
```

Comment:
Currently implemented to only notify if the active difficulty changes - is this correct or should it notify on every iteration even if active difficulty remains the same?

Reply:
Only when it changes I think

Reply:
If the call is sufficiently lightweight, I could see value in triggering it every cycle, specifically to ensure consistent data points for any tracking/visualizations, and also as a mechanism that allows watching for that consistency. If gaps are seen, this allows polling for difficulty separately through RPC as fallback or alerting of a possible issue (if these notifications aren't seen for a time period). Any additional thoughts @cryptocode or @guilhermelawless ?

Reply:
I think that's a good idea, it's very light data especially considering the ~20 second period.

Reply:
Sounds good to me if it simplifies clients and is low frequency.

Reply:
The `request_loop` runs faster in testing - will publishing on every iteration have an impact on other tests if running faster?:

`request_interval_ms = is_test_network () ? (is_sanitizer_build ? 100 : 20) : 16000;`

Reply:
Not a problem I believe, there is much higher frequency data being passed around. Tests should pass.

Reply:
We agree with @guilhermelawless, sending it every cycle should be fine.